NExtends.Tests/CultureContextTest.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using NExtends.Context;
+using Xunit;
+
+namespace NExtends.Tests
+{
+    public class CultureContextTest
+    {
+        [Fact]
+        public void SimpleSwapString()
+        {
+            using (new CultureContext("fr-FR"))
+            {
+                Assert.Equal(new CultureInfo("fr-FR"), CultureInfo.CurrentCulture);
+                using (new CultureContext("en-EN"))
+                {
+                    Assert.Equal(new CultureInfo("en-EN"), CultureInfo.CurrentCulture);
+                }
+                Assert.Equal(new CultureInfo("fr-FR"), CultureInfo.CurrentCulture);
+            }
+        }
+
+        [Fact]
+        public void SimpleSwapCultureInfo()
+        {
+            var fr = new CultureInfo("fr-FR");
+            var en = new CultureInfo("en-EN");
+            using (new CultureContext(fr))
+            {
+                Assert.Equal(fr, CultureInfo.CurrentCulture);
+                using (new CultureContext(en))
+                {
+                    Assert.Equal(en, CultureInfo.CurrentCulture);
+                }
+                Assert.Equal(fr, CultureInfo.CurrentCulture);
+            }
+        }
```

Comment:
Here you just test that CultureInfo is not Equal by reference don't you ?!

So you can thrash this test and add a test that assert that cultureInfo is Equal by value

Reply:
CultureInfo overrides equals operator : https://referencesource.microsoft.com/#mscorlib/system/globalization/cultureinfo.cs,3c87ed6650eea1ca,references